### PR TITLE
Error states

### LINKED ISSUE #106: Show appropriate outputs for different errors
For example, if there is an internet issue, say that cannot connect to the internet - does not say so. Similarly, handle other error codes gracefully.

### PR DESCRIPTION
**Pull request by deepansh96**
Fixes #106 

## Summary
- [x] Added a package [vue-toastification](https://github.com/Maronato/vue-toastification/tree/next) and replaced existing toast implementation with this
- [x] Added a toast which will pop up if the user enters a wrong OTP
- [x] For any errors that are caught while doing google auth, a toast saying "Something went wrong. Please try again" will popup
- [x] All responses returned by the API calls will be intercepted by an interceptor and network errors will be handled -- a toast saying "A network error occurred"
- [x] Added a generic file ErrorHandling.js in `services/Functional/` where the network error state handling is present. More types of generic error states can be specified there

## Test Plan
tested locally

### CODE REVIEW

File: src/pages/Player.vue [hunk 1]
```diff
@@ -179,6 +180,8 @@ export default {
       sessionDBId: null, // id for this session in the plio DB table
       retention: [], // array to store video retention value
       lastTimestampRetention: null, // last recorded timestamp in the retention array
+      toast: useToast(), // use the toast component
+      somethingWentWrongWarning: "Something went wrong. Please try again!",
```
Comment (dalmia) [inline src/pages/Player.vue:184]:
```suggestion
      genericWarning: "Something went wrong. Please try again!",
```

File: src/components/UI/Text/URL.vue [hunk 2]
```diff
@@ -46,9 +45,7 @@ export default {
       type: [String, Object],
     },
   },
-  components: {
-    Toast,
-  },
+  components: {},
```
Comment (dalmia) [inline src/components/UI/Text/URL.vue:48]:
```suggestion
```

File: src/services/Functional/ErrorHandling.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { useToast } from "vue-toastification";
+const toast = useToast();
+
+export default {
+  handleAPIErrors(error) {
+    // TODO - Add more types of errors here
+    if (error.message == "Network Error")
+      toast.error("A network error occurred.");
```
Comment (dalmia) [inline src/services/Functional/ErrorHandling.js:8]:
```suggestion
      toast.error("Please check your internet connection");
```

File: src/services/Functional/ErrorHandling.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { useToast } from "vue-toastification";
+const toast = useToast();
+
+export default {
+  handleAPIErrors(error) {
+    // TODO - Add more types of errors here
```
Comment (dalmia) [inline src/services/Functional/ErrorHandling.js:6]: don't think we need this here

File: src/services/Functional/ErrorHandling.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    if (error.message == "Network Error")
```
Comment (dalmia) [inline src/services/Functional/ErrorHandling.js:7]: can't we do it using response code? is the message always going to be this?

File: src/services/API/RootClient.js [hunk 6]
```diff
@@ -34,6 +35,18 @@ client.interceptors.request.use(
...
+  }
+);
+
+// handle errors in responses for API requests
+client.interceptors.response.use(
+  (config) => {
+    return config;
+  },
+  (error) => {
+    ErrorHandling.handleAPIErrors(error);
```
Comment (dalmia) [inline src/services/API/RootClient.js:49]: Are you sure both the types of errors should be handled by the same thing? The request hasn't happened only in the upper one - so, there won't be any API errors no? the error there is just any error in the interceptor code right?

File: src/pages/Login.vue [hunk 7]
```diff
@@ -98,6 +99,9 @@ export default {
       requestedOtp: false, // whether the user has requested OTP once
       resentOtp: false, // whether the user has requested to resend OTP
       invalidOtp: false, // whether the OTP is invalid
+      toast: useToast(), // use the toast component
+      wrongOTPWarning: "Incorrect OTP entered. Please try again!", // Warning to show when entered OTP is incorrect
+      somethingWentWrongWarning: "Something went wrong. Please try again!",
```
Comment (dalmia) [inline src/pages/Login.vue:104]:
```suggestion
      generic: "Something went wrong. Please try again!",

```

File: src/services/Functional/ErrorHandling.js (hunk 5 above)
Comment (deepansh96) [reply src/services/Functional/ErrorHandling.js:7]: https://github.com/axios/axios/issues/383

File: src/services/Functional/ErrorHandling.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { useToast } from "vue-toastification";
+const toast = useToast();
+
+export default {
+  handleAPIErrors(error) {
+    if (!error.response) {
+      // https://github.com/axios/axios/issues/383
```
Comment (deepansh96) [inline src/services/Functional/ErrorHandling.js:7]:
```suggestion
```

**Pull request merged**